pandas/tests/generic/test_frame.py

```diff
@@ -287,3 +289,36 @@ def test_deepcopy_empty(self):
         empty_frame_copy = deepcopy(empty_frame)
 
         self._compare(empty_frame_copy, empty_frame)
+
+    def test_datetimeField_after_setitem(self):
+        # This test covers the unexpected behaviour of datetimeField when using
+        # setitem on another column as reported in issue #6942
+
+        start = pd.to_datetime("20140401")
+
+        df = pd.DataFrame(
+            index=pd.date_range(start, periods=1), columns=["timenow", "Live"]
+        )
+
+        df.at[start, "timenow"] = datetime.today()  # initial time.
+        time1 = df.at[start, "timenow"]
+
+        time.sleep(1)  # sleep time of 1 second in between assignments.
+
+        df.at[
+            start, "timenow"
+        ] = datetime.today()  # modified time before 'Live' column is set.
+        time2 = df.at[start, "timenow"]
+
+        df.Live = True  # setting the 'Live' column to True.
+
+        time.sleep(1)  # sleep time of 1 second in between assignments.
+
+        df.at[
+            start, "timenow"
+        ] = datetime.today()  # modified time after 'Live' column is set.
+        time3 = df.at[start, "timenow"]
+
+        assert not time1 == time2
```

Comment:
We would like to construct 2 dataframes here and use `assert_frame_equal` to compare them

```
result = ...
expected = ...
tm.assert_frame_equal(result, expected)
```

Reply:
I thought of the same but the whole point of the test is to make sure that the datetimeField changes with each assignment and as such would require to assert that frames are not equal, that's went with this. Couldn't find a more elegant way round this.